force-app/main/default/classes/SOQL.cls

```diff
@@ -108,16 +126,28 @@ public with sharing class SOQL {
      * @return A list of Opportunity records.
      */
     public static List<Opportunity> getOpportunitiesByNameKeyword(String keyword) {
-        return null; // Replace null with the variable you used to store the result
+        if(keyword == null){
+            return new List <Opportunity>();
+        }
+        List <Opportunity> opps = [SELECT Id, Name FROM Opportunity];
+        List <Opportunity> oppsWithKeyword = new List <Opportunity>();
+        for(Opportunity o : opps){
+            if(o.Name.contains(keyword)){
+                oppsWithKeyword.add(o);
+            }
+        }
+        return oppsWithKeyword; // Replace null with the variable you used to store the result
```

Comment:
Great logic here, however, we can do this filter in the query. Use the LIKE Operator with the keyword and some wildcards '%'. Then you don't have to iterate, you can just query WHERE Name LIKE :keyword and you will be all set.